pkgs/applications/networking/p2p/peergos/default.nix

```
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchurl, jre, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  pname = "peergos";
+  version = "0.14.1";
+
+  peergos = fetchurl {
+    url = "https://github.com/Peergos/web-ui/releases/download/v${version}/Peergos.jar";
+    hash = "sha256-oCsUuFxTAL0vAabGggGhZHaF40A5TLfkT15HYPiKHlU=";
+  };
+
+  dontUnpack = 1;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    mkdir -p $out/share/java/
+    cp $peergos $out/share/java/peergos.jar
+
+    makeWrapper ${jre}/bin/java $out/bin/peergos --chdir $out/share/peergos/ --add-flags "-Djava.library.path=$out/share/peergos/native-lib/libtweetnacl.so -jar $out/share/java/peergos.jar"
```

```suggestion
    makeWrapper ${lib.getExe jre} $out/bin/peergos \
      --chdir $out/share/peergos/ \
      --add-flags "-Djava.library.path=$out/share/peergos/native-lib/libtweetnacl.so -jar $out/share/java/peergos.jar"
```
Why is chdir needed?

---

It's because running `peergos` creates folder _native-lib/libtweetnacl.so_ in current `pwd`

---

You should remove it, see below

---

Can we pass an environment variable or do to change the location of the file?

---

I don't think so, it seems to be a bug. I was trying to do that by setting `-Djava.library.path` but it has no effect.

---

In that case please open an upstream issue/file a PR.